app/api/v1/endpoints/weather.py

```
@@ -38,37 +39,46 @@ async def fetch_weather(
+            weather_request.longitude,
+            entry_datetime_utc,
         )
 
         if not weather_data:
+             # This happens if parsing fails or data is missing despite success status
+             # We treat it as service unavailable or data unavailable
             raise HTTPException(
                 status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
-                detail="Failed to parse weather data"
+                detail="Failed to retrieve weather data from provider."
             )
```

Ruff (0.14.10)

Abstract `raise` to an inner function

(TRY301)